### PR TITLE
refac: refactor function names and parameters in repository contracts 

### LINKED ISSUE #16: Remover nome de parâmetros nos contratos (interfaces)

### LINKED ISSUE #29: Mudar funções Get por GetByExternalID

### PR DESCRIPTION
**Pull request by eakeur**
This pull request intends to refactor repository interfaces so that the code gets more readable. These were the main changes till now: 

- **Get** functions on **account.Repository** and **transfer.Repository** now are named **GetByExternalID** to expose the right key to be parameterized

- **Create** functions used to receive a pointer to (Ex.) **Account**. This behavior existed so that the user of the function would have the same object with updated fields, but now the same behavior is achieved by also returning an (Ex.) **Account** from the function. 

- These changes mentioned above sure affected many other parts of the written code, such as tests. So most of the files changed in this PR are actually side effects fixes, not exactly any change to the way the app works.

closes #16 
closes #29

### CODE REVIEW

**Comment by xeniaka** [comment]
> This pull request intends to refactor repository interfaces so that the code gets more readable. These were the main changes till now:
> 
> * **Get** functions on **account.Repository** and **transfer.Repository** now are named **GetByExternalID** to expose the right key to be parameterized
> * **Create** functions used to receive a pointer to (Ex.) **Account**. This behavior existed so that the user of the function would have the same object with updated fields, but now the same behavior is achieved by also returning an (Ex.) **Account** from the function.
> * These changes mentioned above sure affected many other parts of the written code, such as tests. So most of the files changed in this PR are actually side effects fixes, not exactly any change to the way the app works.
> 
> closes #16 closes #29

Ah the descriptions... ❤️

File: app/gateway/database/postgres/account/create.go [hunk 1]
```diff
@@ -3,14 +3,13 @@ package account
...
-	"stonehenge/app/core/types/id"
 	"stonehenge/app/gateway/database/postgres/common"
 )
 
-func (r *repository) Create(ctx context.Context, acc *account.Account) (id.ExternalID, error) {
+func (r *repository) Create(ctx context.Context, acc account.Account) (account.Account, error) {
 	db, found := common.TransactionFrom(ctx)
 	if !found {
-		return id.New(), account.ErrCreating
+		return acc, account.ErrCreating
```
Comment (xeniaka) [inline app/gateway/database/postgres/account/create.go:12]: We usually return an empty struct in our projects.

```suggestion
		return account.Account{}, account.ErrCreating
```

Comment (xeniaka) [reply app/gateway/database/postgres/account/create.go:12]: Also, if you return a custom error, whoever is using the function is going to miss valuable information.
You may just return the error the database returned and let the other layers deal with it.

Comment (eakeur) [reply app/gateway/database/postgres/account/create.go:12]: Oh, so it's ok to let a DB error leave its layer, for example?

**Pull request merged**